### PR TITLE
Implement StoreVectorAndZipNxM for Arm64

### PR DESCRIPTION
**Pull request by SwapnilGaikwad**
Contribute towards https://github.com/dotnet/runtime/issues/84510.

Implements the StoreVectorAndZipNxM for Arm64 N=(64,128) & M=(2, 3, 4) API calls

```c#
public static unsafe void StoreVector64x2AndZip(byte*   address, (Vector64<byte>   Value1, Vector64<byte>   Value2) value);
public static unsafe void StoreVector64x2AndZip(sbyte*  address, (Vector64<sbyte>  Value1, Vector64<sbyte>  Value2) value);
public static unsafe void StoreVector64x2AndZip(short*  address, (Vector64<short>  Value1, Vector64<short>  Value2) value);
public static unsafe void StoreVector64x2AndZip(ushort* address, (Vector64<ushort> Value1, Vector64<ushort> Value2) value);
public static unsafe void StoreVector64x2AndZip(int*    address, (Vector64<int>    Value1, Vector64<int>    Value2) value);
public static unsafe void StoreVector64x2AndZip(uint*   address, (Vector64<uint>   Value1, Vector64<uint>   Value2) value);
public static unsafe void StoreVector64x2AndZip(float*  address, (Vector64<float>  Value1, Vector64<float>  Value2) value);

public static unsafe void StoreVector64x3AndZip(byte*   address, (Vector64<byte>   Value1, Vector64<byte>   Value2, Vector64<byte>   Value3) value);
public static unsafe void StoreVector64x3AndZip(sbyte*  address, (Vector64<sbyte>  Value1, Vector64<sbyte>  Value2, Vector64<sbyte>  Value3) value);
public static unsafe void StoreVector64x3AndZip(short*  address, (Vector64<short>  Value1, Vector64<short>  Value2, Vector64<short>  Value3) value);
public static unsafe void StoreVector64x3AndZip(ushort* address, (Vector64<ushort> Value1, Vector64<ushort> Value2, Vector64<ushort> Value3) value);
public static unsafe void StoreVector64x3AndZip(int*    address, (Vector64<int>    Value1, Vector64<int>    Value2, Vector64<int>    Value3) value);
public static unsafe void StoreVector64x3AndZip(uint*   address, (Vector64<uint>   Value1, Vector64<uint>   Value2, Vector64<uint>   Value3) value);
public static unsafe void StoreVector64x3AndZip(float*  address, (Vector64<float>  Value1, Vector64<float>  Value2, Vector64<float>  Value3) value);

public static unsafe void StoreVector64x4AndZip(byte*   address, (Vector64<byte>   Value1, Vector64<byte>   Value2, Vector64<byte>   Value3, Vector64<byte>   Value4) value);
public static unsafe void StoreVector64x4AndZip(sbyte*  address, (Vector64<sbyte>  Value1, Vector64<sbyte>  Value2, Vector64<sbyte>  Value3, Vector64<sbyte>  Value4) value);
public static unsafe void StoreVector64x4AndZip(short*  address, (Vector64<short>  Value1, Vector64<short>  Value2, Vector64<short>  Value3, Vector64<short>  Value4) value);
public static unsafe void StoreVector64x4AndZip(ushort* address, (Vector64<ushort> Value1, Vector64<ushort> Value2, Vector64<ushort> Value3, Vector64<ushort> Value4) value);
public static unsafe void StoreVector64x4AndZip(int*    address, (Vector64<int>    Value1, Vector64<int>    Value2, Vector64<int>    Value3, Vector64<int>    Value4) value);
public static unsafe void StoreVector64x4AndZip(uint*   address, (Vector64<uint>   Value1, Vector64<uint>   Value2, Vector64<uint>   Value3, Vector64<uint>   Value4) value);
public static unsafe void StoreVector64x4AndZip(float*  address, (Vector64<float>  Value1, Vector64<float>  Value2, Vector64<float>  Value3, Vector64<float>  Value4) value);


public static unsafe void StoreVector128x2AndZip(byte*   address, (Vector128<byte>   Value1, Vector128<byte>   Value2) value);
public static unsafe void StoreVector128x2AndZip(sbyte*  address, (Vector128<sbyte>  Value1, Vector128<sbyte>  Value2) value);
public static unsafe void StoreVector128x2AndZip(short*  address, (Vector128<short>  Value1, Vector128<short>  Value2) value);
public static unsafe void StoreVector128x2AndZip(ushort* address, (Vector128<ushort> Value1, Vector128<ushort> Value2) value);
public static unsafe void StoreVector128x2AndZip(int*    address, (Vector128<int>    Value1, Vector128<int>    Value2) value);
public static unsafe void StoreVector128x2AndZip(uint*   address, (Vector128<uint>   Value1, Vector128<uint>   Value2) value);
public static unsafe void StoreVector128x2AndZip(long*   address, (Vector128<long>   Value1, Vector128<long>   Value2) value);
public static unsafe void StoreVector128x2AndZip(ulong*  address, (Vector128<ulong>  Value1, Vector128<ulong>  Value2) value);
public static unsafe void StoreVector128x2AndZip(float*  address, (Vector128<float>  Value1, Vector128<float>  Value2) value);
public static unsafe void StoreVector128x2AndZip(double* address, (Vector128<double> Value1, Vector128<double> Value2) value);

public static unsafe void StoreVector128x3AndZip(byte*   address, (Vector128<byte>   Value1, Vector128<byte>   Value2, Vector128<byte>   Value3) value);
public static unsafe void StoreVector128x3AndZip(sbyte*  address, (Vector128<sbyte>  Value1, Vector128<sbyte>  Value2, Vector128<sbyte>  Value3) value);
public static unsafe void StoreVector128x3AndZip(short*  address, (Vector128<short>  Value1, Vector128<short>  Value2, Vector128<short>  Value3) value);
public static unsafe void StoreVector128x3AndZip(ushort* address, (Vector128<ushort> Value1, Vector128<ushort> Value2, Vector128<ushort> Value3) value);
public static unsafe void StoreVector128x3AndZip(int*    address, (Vector128<int>    Value1, Vector128<int>    Value2, Vector128<int>    Value3) value);
public static unsafe void StoreVector128x3AndZip(uint*   address, (Vector128<uint>   Value1, Vector128<uint>   Value2, Vector128<uint>   Value3) value);
public static unsafe void StoreVector128x3AndZip(long*   address, (Vector128<long>   Value1, Vector128<long>   Value2, Vector128<long>   Value3) value);
public static unsafe void StoreVector128x3AndZip(ulong*  address, (Vector128<ulong>  Value1, Vector128<ulong>  Value2, Vector128<ulong>  Value3) value);
public static unsafe void StoreVector128x3AndZip(float*  address, (Vector128<float>  Value1, Vector128<float>  Value2, Vector128<float>  Value3) value);
public static unsafe void StoreVector128x3AndZip(double* address, (Vector128<double> Value1, Vector128<double> Value2, Vector128<double> Value3) value);

public static unsafe void StoreVector128x4AndZip(byte*   address, (Vector128<byte>   Value1, Vector128<byte>   Value2, Vector128<byte>   Value3, Vector128<byte>   Value4) value);
public static unsafe void StoreVector128x4AndZip(sbyte*  address, (Vector128<sbyte>  Value1, Vector128<sbyte>  Value2, Vector128<sbyte>  Value3, Vector128<sbyte>  Value4) value);
public static unsafe void StoreVector128x4AndZip(short*  address, (Vector128<short>  Value1, Vector128<short>  Value2, Vector128<short>  Value3, Vector128<short>  Value4) value);
public static unsafe void StoreVector128x4AndZip(ushort* address, (Vector128<ushort> Value1, Vector128<ushort> Value2, Vector128<ushort> Value3, Vector128<ushort> Value4) value);
public static unsafe void StoreVector128x4AndZip(int*    address, (Vector128<int>    Value1, Vector128<int>    Value2, Vector128<int>    Value3, Vector128<int>    Value4) value);
public static unsafe void StoreVector128x4AndZip(uint*   address, (Vector128<uint>   Value1, Vector128<uint>   Value2, Vector128<uint>   Value3, Vector128<uint>   Value4) value);
public static unsafe void StoreVector128x4AndZip(long*   address, (Vector128<long>   Value1, Vector128<long>   Value2, Vector128<long>   Value3, Vector128<long>   Value4) value);
public static unsafe void StoreVector128x4AndZip(ulong*  address, (Vector128<ulong>  Value1, Vector128<ulong>  Value2, Vector128<ulong>  Value3, Vector128<ulong>  Value4) value);
public static unsafe void StoreVector128x4AndZip(float*  address, (Vector128<float>  Value1, Vector128<float>  Value2, Vector128<float>  Value3, Vector128<float>  Value4) value);
public static unsafe void StoreVector128x4AndZip(double* address, (Vector128<double> Value1, Vector128<double> Value2, Vector128<double> Value3, Vector128<double> Value4) value);

```

### CODE REVIEW

**Comment by SwapnilGaikwad** [comment]
Hi @kunalspathak , these are the remaining StoreVector* APIs, mostly boilerplate code 🙂.

File: src/coreclr/jit/hwintrinsiccodegenarm64.cpp [hunk 1]
```diff
@@ -796,6 +796,12 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                 GetEmitter()->emitIns_R_R_R(ins, emitTypeSize(intrin.baseType), op2Reg, op3Reg, op1Reg);
                 break;
 
+            case NI_AdvSimd_StoreVector64x2AndZip:
+            case NI_AdvSimd_StoreVector64x3AndZip:
+            case NI_AdvSimd_StoreVector64x4AndZip:
+            case NI_AdvSimd_Arm64_StoreVector128x2AndZip:
+            case NI_AdvSimd_Arm64_StoreVector128x3AndZip:
+            case NI_AdvSimd_Arm64_StoreVector128x4AndZip:
```
Comment (kunalspathak) [inline src/coreclr/jit/hwintrinsiccodegenarm64.cpp:804]: looks like we should update the assert:

```
 assert((ins == INS_st1_2regs && regCount == 2) || (ins == INS_st1_3regs && regCount == 3) ||
                       (ins == INS_st1_4regs && regCount == 4));
```

Comment (SwapnilGaikwad) [reply src/coreclr/jit/hwintrinsiccodegenarm64.cpp:804]: Oops! Done.

**Review by kunalspathak — APPROVED** [review]
LGTM

**Pull request merged**